metron-interface/metron-rest/src/main/java/org/apache/metron/rest/config/KafkaConfig.java

```diff
@@ -108,6 +108,9 @@ public ZkUtils zkUtils() {
     producerConfig.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
     producerConfig.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
     producerConfig.put("request.required.acks", 1);
+    if (environment.getProperty(MetronRestConstants.KERBEROS_ENABLED_SPRING_PROPERTY, Boolean.class, false)) {
+      producerConfig.put("security.protocol", "SASL_PLAINTEXT");
```

Comment:
Shouldn't we pull this from the `KAFKA_SECURITY_PROTOCOL` environment variable?  Looks like it's being set in `metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/templates/metron.j2`

Reply:
Is the KAFKA_SECURITY_PROTOCOL env variable guaranteed to be available on the the host where REST is installed?  Either way I don't have a problem with making the Kafka security protocol configurable vs hard-coded.  I would propose we manage it like the other REST properties though and put it in `metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/templates/metron.j2` to ensure it's always available to REST.  Would it make sense to remove the "kerberos.enabled" spring property in this case?

Reply:
Well, if `metron.j2` were not available as environment variables for REST, how does `metron-rest.sh` get the spring profile to use (the `METRON_SPRING_PROFILES_ACTIVE` variable)?

Reply:
`metron.j2` is always available to REST.  KAFKA_SECURITY_PROTOCOL env variable may or may not be depending on if a Kafka Broker or Client is located on the same host (am I wrong?).

Reply:
KAFKA_SECURITY_PROTOCOL is carried through by metron.j2 (`KAFKA_SECURITY_PROTOCOL="{{kafka_security_protocol}}"`).  Given that, we should be fine to use it here, I believe.

Reply:
I submitted a [PR](https://github.com/merrimanr/incubator-metron/pull/35) against your branch with my proposed solution to this.  It fixes the issue for both the producer and consumer configs.  If you choose to accept it, you can do it without my attribution.

Reply:
Thanks @cestella.  Turns out this setting was already available so it was a simple change.  I also added the default application.yml and adjusted the unit test.